### PR TITLE
self: 4.5.0 -> 2017.1

### PR DESCRIPTION
**Pull request by siraben**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by prusnak — APPROVED** [review]

File: pkgs/development/interpreters/self/default.nix [hunk 1]
```diff
@@ -11,16 +11,18 @@ stdenv.mkDerivation {
   # This image can later be started with:
   #   $ Self -s myimage.snap
   #
-  version = "4.5.0";
+  version = "2017.1";
   pname = "self";
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/self/default.nix:15]:
```suggestion
  pname = "self";
  version = "2017.1";
```

File: pkgs/development/interpreters/self/default.nix [hunk 2]
```diff
@@ -11,16 +11,18 @@ stdenv.mkDerivation {
...
+  src = fetchFromGitHub {
+    owner = "russellallen";
+    repo = pname;
+    rev = version;
+    sha256 = "C/1Q6yFmoXx2F97xuvkm8DxFmmvuBS7uYZOxq/CRNog=";
   };
 
-  buildInputs = [ ncurses xorg.libX11 xorg.libXext makeWrapper cmake ];
+  nativeBuildInputs = [ cmake makeWrapper ];
+  buildInputs = [ ncurses xorg.libX11 xorg.libXext ];
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/self/default.nix:25]:
```suggestion
  buildInputs = [ ncurses libX11 libXext ];
```

File: pkgs/development/interpreters/self/default.nix [hunk 3]
```diff
@@ -1,6 +1,6 @@
-{ fetchgit, stdenv, xorg, makeWrapper, ncurses, cmake }:
+{ stdenv, fetchFromGitHub, xorg, makeWrapper, ncurses, cmake }:
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/self/default.nix:1]:
```suggestion
{ stdenv, fetchFromGitHub, libX11, libXext, makeWrapper, ncurses, cmake }:
```

**Review by prusnak — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review. If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108039` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>self</li>
</ul>
</details>

**Pull request merged**